tests/ImageSharp.Tests/Formats/Png/PngDecoderTests.cs

```diff
@@ -491,6 +492,34 @@ public void Issue410_MalformedApplePng<TPixel>(TestImageProvider<TPixel> provide
             Assert.Contains("Proprietary Apple PNG detected!", ex.Message);
         }
 
+        // https://github.com/SixLabors/ImageSharp/issues/2080
+        [Theory]
+        [WithFile(TestImages.Png.Issue2080, PixelTypes.Rgba32)]
+        public void Issue2080_MemoryLeakIn(TestImageProvider<Rgba32> provider)
+        {
+            static void RunTest(string providerDump)
+            {
+                TestImageProvider<Rgba32> provider
+                    = BasicSerializer.Deserialize<TestImageProvider<Rgba32>>(providerDump);
+
+                provider.Configuration = Configuration.CreateDefaultInstance();
+
+                // Replace memory allocator to bypass image cache.
+                // Allocator must be unmanaged to enable allocation tracing.
+                provider.Configuration.MemoryAllocator = new UniformUnmanagedMemoryPoolMemoryAllocator(null);
+
+                int undisposedAllocationCount = MemoryDiagnostics.TotalUndisposedAllocationCount;
```

Comment:
I'm fairly confident trying to write consuming `MemoryDiagnostics.TotalUndisposedAllocationCount` will produce flaky tests, I believe the tests run in parallel and as such other tests could be running at the same time as this one and thus can also be influencing this counter mid test.

I feel we might need to add something to the MemoryAllocator itself to track stats per Manager to make this workable or we might need to alter the MemoryDiagnostics to allow resetting/tracking via an AsyncLocal to allow isolating stats per test.

Reply:
Test code runs using `RemoteExecutor`, which I believe creates new process to execute code. So it won't be flaky.

Reply:
ah yes your right... I missed the `RemoteExecutor` usage... ignore my concerns I can't see how other tests will influence them. 👍